jeos-firstboot-enroll

```diff
@@ -155,21 +187,61 @@ is_pcr_oracle() {
 		[ -e /etc/systemd/tpm2-pcr-private-key.pem ]
 }
 
+write_issue_file() {
+	if [ -e '/usr/sbin/issue-generator' ] && [ -z "$dry" ]; then
+		mkdir -p "/run/issue.d/"
+		issuefile="/run/issue.d/90-diskencrypt.conf"
+	else
+		issuefile='/dev/stdout'
+	fi
+
+	echo -ne "Encryption recovery key:\n  " > "$issuefile"
+	keyctl pipe "$crypt_keyid" >> "$issuefile"
+	echo -e "\n" >> "$issuefile"
+	if [ -x /usr/bin/qrencode ]; then
+		echo "You can also scan it with your mobile phone:" >> "$issuefile"
+		keyctl pipe "$crypt_keyid" | qrencode -t utf8i >> "$issuefile"
+	fi
+
+	run issue-generator
+	[ -n "$dry" ] || cat "$issuefile"
+}
+
+add_password() {
+	[ -n "$crypt_pw" ] || return 0
+	local dev
+	for dev in "${luks2_devices[@]}"; do
+		echo "adding password to $dev"
+		echo -n "$crypt_pw" | run cryptsetup luksAddKey --verbose --batch-mode --force-password --key-file <(keyctl pipe "$crypt_keyid") "$dev"
```

Comment:
why not to use cryptsetup? you can use the keyring for the keyid one, and a envvar to set the new one

Reply:
for a separate commit then